frontend/app/src/components/dashboard/NftBalanceTable.vue

```diff
@@ -182,18 +198,59 @@ const percentageOfTotalNetValue = (value: BigNumber) => {
 };
 
 const percentageOfCurrentGroup = (value: BigNumber) => {
-  return calculatePercentage(value, get(total));
+  return calculatePercentage(value, get(totalUsdValue) as BigNumber);
 };
 
 const refresh = async () => {
-  return await fetchNonFungibleBalances({ ignoreCache: true });
+  return await fetchNonFungibleBalances(true);
 };
 
 const { dashboardTablesVisibleColumns } = storeToRefs(
   useFrontendSettingsStore()
 );
 
-const filteredBalances = computed(() => {
-  return get(nonFungibleBalances).filter(item => !item.isLp);
+const { data, limit, found, total, totalUsdValue } =
+  getCollectionData<NonFungibleBalance>(
+    balances as Ref<Collection<NonFungibleBalance>>
+  );
+
+const { itemLength } = setupEntryLimit(limit, found, total);
+
+const options: Ref<PaginationOptions | null> = ref(null);
+
+const updatePaginationHandler = async (
+  newOptions: PaginationOptions | null
+) => {
+  set(options, newOptions);
+  await updatePayloadHandler();
+};
+
+const updatePayloadHandler = async () => {
+  let paginationOptions = {};
+
+  const optionsVal = get(options);
+  if (optionsVal) {
+    const { itemsPerPage, page, sortBy, sortDesc } = optionsVal;
+    const offset = (page - 1) * itemsPerPage;
+
+    paginationOptions = {
+      limit: itemsPerPage,
+      offset,
+      orderByAttributes: sortBy.length > 0 ? sortBy : ['name'],
+      ascending:
+        sortDesc.length > 1 ? dropRight(sortDesc).map(bool => !bool) : [true]
```

Comment:
Is there any specific reason why we remove one of the elements when ordering? I remember seeing dropRight used elsewhere but I am unsure why we need it.

Reply:
sometimes something weird happens. The `orderByAttributes` have 1 value, but the `sortDesc` have 2. But I notice the correct sort value is the first one.

Reply:
Ok, this is weird, maybe we should look into why this happens.